scipy/signal/tests/test_signaltools.py

```diff
@@ -267,6 +268,36 @@ def test_fillvalue(self):
                    [32, 46, 67, 62, 48]])
         assert_array_equal(c, d)
 
+    def test_fillvalue_deprecations(self):
+        # Deprecated 2017-07, scipy version 1.0.0
+        with suppress_warnings() as sup:
+            sup.filter(np.ComplexWarning, "Casting complex values to real")
+            r = sup.record(DeprecationWarning, "could not cast `fillvalue`")
+            convolve2d([[1]], [[1, 2]], fillvalue=1j)
+            assert_(len(r) == 1)
```

Comment:
Had to avoid assert_warns, works on newer numpy version which is more specific with suppress_warnings, but not on older.